tests/unit_tests/test_embedding_function_registry.py

```
@@ -0,0 +1,432 @@
+        EmbeddingFunctionRegistry.register(TestEmbeddingFunction)
+
+        # Verify it's registered
+        assert "test_embedding" in EmbeddingFunctionRegistry.list_registered()
+        cls = EmbeddingFunctionRegistry.get_class("test_embedding")
+        assert cls == TestEmbeddingFunction
+
+    def test_register_missing_name_method(self):
+        """Test that registering a class without name() method raises ValueError"""
+
+        class InvalidEmbeddingFunction(EmbeddingFunction[Documents]):
+            def __call__(self, input: Documents) -> Embeddings:
```

Ruff (0.14.13)

Unused method argument: `input`

(ARG002)